src/detect-tls-random.c

```diff
@@ -0,0 +1,275 @@
+            GetRandomData, ALPROTO_TLS, 0);
+
+    /* Register engine for Client random */
+    DetectAppLayerInspectEngineRegister2("tls.random", ALPROTO_TLS, SIG_FLAG_TOCLIENT, 0,
+            DetectEngineInspectBufferGeneric, GetRandomData);
+    DetectAppLayerMpmRegister2("tls.random", SIG_FLAG_TOCLIENT, 2, PrefilterGenericMpmRegister,
+            GetRandomData, ALPROTO_TLS, 0);
+
+    DetectBufferTypeSetDescriptionByName("tls.random", "TLS Random");
+
+    g_tls_random_buffer_id = DetectBufferTypeGetByName("tls.random");
+}
+
+/**
+ * \brief this function setup the tls.random_time sticky buffer keyword used in the rule
+ *
+ * \param de_ctx   Pointer to the Detection Engine Context
+ * \param s        Pointer to the Signature to which the current keyword belongs
+ * \param str      Should hold an empty string always
+ *
+ * \retval 0  On success
+ * \retval -1 On failure
+ */
+static int DetectTlsRandomTimeSetup(DetectEngineCtx *de_ctx, Signature *s, const char *str)
+{
+    if (DetectBufferSetActiveList(s, g_tls_random_time_buffer_id) < 0)
+        return -1;
+
+    if (DetectSignatureSetAppProto(s, ALPROTO_TLS) < 0)
+        return -1;
+
+    return 0;
+}
+
+/**
+ * \brief this function setup the tls.random_bytes sticky buffer keyword used in the rule
+ *
+ * \param de_ctx   Pointer to the Detection Engine Context
+ * \param s        Pointer to the Signature to which the current keyword belongs
+ * \param str      Should hold an empty string always
+ *
+ * \retval 0  On success
+ * \retval -1 On failure
+ */
+static int DetectTlsRandomBytesSetup(DetectEngineCtx *de_ctx, Signature *s, const char *str)
+{
+    if (DetectBufferSetActiveList(s, g_tls_random_bytes_buffer_id) < 0)
+        return -1;
+
+    if (DetectSignatureSetAppProto(s, ALPROTO_TLS) < 0)
+        return -1;
+
+    return 0;
+}
+
+/**
+ * \brief this function setup the tls.random sticky buffer keyword used in the rule
+ *
+ * \param de_ctx   Pointer to the Detection Engine Context
+ * \param s        Pointer to the Signature to which the current keyword belongs
+ * \param str      Should hold an empty string always
+ *
+ * \retval 0  On success
+ * \retval -1 On failure
+ */
+static int DetectTlsRandomSetup(DetectEngineCtx *de_ctx, Signature *s, const char *str)
+{
+    if (DetectBufferSetActiveList(s, g_tls_random_buffer_id) < 0)
+        return -1;
+
+    if (DetectSignatureSetAppProto(s, ALPROTO_TLS) < 0)
+        return -1;
+
+    return 0;
+}
+
+static InspectionBuffer *GetRandomTimeData(DetectEngineThreadCtx *det_ctx,
+        const DetectEngineTransforms *transforms, Flow *f, const uint8_t flow_flags, void *txv,
+        const int list_id)
+{
+    InspectionBuffer *buffer = InspectionBufferGet(det_ctx, list_id);
+    if (buffer->inspect == NULL) {
+        const SSLState *ssl_state = (SSLState *)f->alstate;
+        if (((flow_flags & STREAM_TOSERVER) && !ssl_state->server_connp.random_set) ||
+                !ssl_state->client_connp.random_set) {
```

Comment:
I find logic this hard to read.

Reply:
Which one of the following do you think?
1. Add comment explaining what's happening.
2. More if..else blocks with readable logic e.g. https://github.com/OISF/suricata/pull/7754/files#diff-5cedc2be711264b96bff6f33bfef99c5f2de6f6616ab397f5c842a933201b97cR252

Reply:
I prefer readable code. The compiler will turn it into the same compiled code most likely, but us pesky humans can understand it much quicker.